src/reconnection.ts

```diff
@@ -152,11 +151,12 @@ export class ReconnectManager<PeerMetadata, TrackMetadata> {
     }, timeout);
   }
 
-  private handleReconnect() {
-    if (!this.ongoingReconnection) return;
+  public async handleReconnect() {
+    if (this.status !== 'RECONNECTING') return;
 
     if (this.lastLocalEndpoint && this.reconnectConfig.addTracksOnReconnect) {
-      this.lastLocalEndpoint.tracks.forEach(async (track) => {
+      for await (const element of this.lastLocalEndpoint.tracks) {
+        const [_, track] = element;
```

Comment:
I would like to discuss this part of code, as I would like to understand something.